### PR TITLE
Modify git tests so they continue to work with git < 2.28

### PR DESCRIPTION
**Pull request by ajkavanagh**
git 2.28 introduced the `--initial-branch` option to
`git init` which allowed the initial branch to be other than
master.  This was used in the GitUrlFetchHandlerTest class.
Unfortunately, this breaks when testing on a distro that doesn't
have git < 2.28 installed.  Modify the tests so that this is called
out explicitly.  When git pre-2.28 is gone, this patch can be
reverted.

### CODE REVIEW

File: tests/fetch/test_giturl.py [hunk 1]
```diff
@@ -36,6 +37,22 @@ def setUp(self):
             "garbage",
         )
         self.fh = giturl.GitUrlFetchHandler()
+        self._git_version_2_28_plus = None
```
Comment (addyess) [inline tests/fetch/test_giturl.py:40]: sigh -- this is why `@cached_property` should have been a thing long before `python3.8`

Comment (ajkavanagh) [reply tests/fetch/test_giturl.py:40]: Indeed.  C-h will bump major version nums as each py3.x goes EOL so that eventually we can use the newer features.

**Review by addyess — APPROVED** [review]
You got my +1

**Review by wolsen — APPROVED** [review]
LGTM

**Pull request merged**